### ISSUE TITLE
Expense pipeline management

### ISSUE THREAD

**Issue by iamronen** [opencollective/opencollective#6790]
## User story 

> As a fiscal host admin paying expenses is at the heart of the services I provide via Open Collective. Collectives expect a reliable and responsive service from me. It is important for me to know when expenses require my attention so that I can tend to them. Currently:

> 1. I regularly pay expenses many of which require inquiry and followup. I use the statuses Ready-to-Pay, On Hold, Incomplete and Re-approval requested to keep track of where expenses are. These “statuses” are a method of organization for me to deal with the work of paying expenses. I currently rely on filtering expenses to get to these expenses but this is tedious , it does not give me an overview of the work I need to do and it makes me feel like a swimming in an endless see of expenses. 

> 2. I am sent emails that I mostly ignore due to email overwhelm. As a result there is now way for me to know that there are expenses that require my attention. I have to pro-actively filter the expense list for different statuses in order to determine what I need to do.

>3. Expenses also trigger (both when they are approved and when there is back and forth communication around incomplete expenses) Freshdesk issues. Most of these are of no value to me, they unnecessarily overload the Freshdesk pipeline and I am left with the additional work (after paying expenses) of having to cleanup expense related tickets on Freshdesk. 

> 4. I pay expenses at more than one fiscal host and/or have to also approve expenses as a collective admin on other collectives. This means that I need to regularly move between different places on the platform to find what it is that I need to do.

### Part 1: Paying Expenses as a “Job To Be Done” (vs "filtering expenses").

Provide fiscal host admins with a tool designed explicitly for managing the expense pipeline:

1. Focus on expenses that are in the pipeline (that require admin attention). Other expenses (completed, rejected, spam, etc.) do not show up here.
2. Build it as a “queue that needs to be emptied” experience. As expenses are submitted the queue fills up, as expenses are processed the queue empties. This is different from the primary (existing) expense list which grows forever.
3. Provide a user experience that gives an overview of the work ahead and gives a sense of progress as expenses are processed (expenses that have, for example, been marked as incomplete, are, from the point of view of a fiscal host admin, and therefore also in this tool, are counted as “done” until something happens to them that places them back in the queue that re-calls the attention of a fiscal host admin).  
4. Adapt the list of expenses to the state the admin is tending to:
  1. When reviewing expenses that have been placed on hold they are displayed the reason an expense was placed on hold. 
  2. When reviewing expenses that have been marked as incomplete they are displayed the reason an expense was marked as incomplete (and can therefore know what to look for or what to expect when engaging with it). 
  3. When reviewing expenses for which re-approval was requested they can see the reason re-approval was requested.
  4. When reviewing expenses for which payment-errors occurred, they can see information about the error to help them to determine what needs to be done next.
  5. etc.

![image](https://github.com/opencollective/opencollective/assets/8337819/cf52e595-9680-4b69-be62-5f9b115624f5)

### Part 2: Calling for Attention (In-app Notifications)

Provide UI mechanism that call out to the use to indicate that their attention is needed.

This project is focused on fiscal host admins and so requires:

1. You have (n) expenses to pay
2. You have (n) payment errors that need to be reviewed.

However, this mechanism should be thought of more broadly since other user roles evoke other notifications, such as:

  1. You have (n) expenses to approve (collective admin)
  2. You have (n) expenses to revise (expense submitter)
  3. You have a payment method that is expiring and needs to be updated.
  4. You have a expenses made with virtual cards that are missing receipts.

This notifications mechanism should:

1. Reliably get the users attention with a distinct visual language and UX patterns that are consistent across the platform (eg: in the “my account control” / workspace sidebars).
2. Effectively communicate to the user what is needed.
3. Redirect the user to wherever they need to go on the platform in order to do the work they need to do.
4. Accommodate situations in which an individual user may have numerous roles on the platform: a fiscal host admin (in more than one host), a collective admin (in more than one collective) and an expense submitter. The mechanism needs to be able to effectively communicate different notification for different tasks in different roles.

![image](https://github.com/opencollective/opencollective/assets/8337819/9c2a1d99-f8eb-4100-9954-e4c79ae85e5c)

## Traps

1. Tool Placement
    1. If the two elements (Paying Expenses tool & In-App Notifications) are addressed well we have flexibility about where the tool lives.  
    2. Placing the tool in the fiscal host admin can remove the added complexity of filtering by fiscal host in the individual workspace. This relies on there being a clear and reliable notification.
    3. If paying expenses is in the fiscal host admin and approving expenses is in the collective admin and entering receipts for virtual card payments in the personal workspace  does this lead to a sense of “hopping around the platform” or are these usecases separated enough (by time and context)?
2. Counters needs to be implemented with consideration to what the user needs to see and experience
     1. The things that “I the fiscal host admin” need to do is to handle ready-to-pay and on-hold expenses.
     2. “I the fiscal host admin” have already handled expenses that are marked as incomplete or requested re-approval. These should not count as things that need to do.
     3. Payment errors (that are caused by the payment processors) are something that “I the fiscal host admin” needs to do, but if they are counted with the things I need to do that would create a less precise and more depressing (and less effective) experience (especially given the current backlog states) of ‘getting work done’. 
4. Integration:
    1. Expenses: how does all this come together so that fiscal host admins can lean into “paying expenses” on the platform, including back and forth communications with expense submitters (and collective admins when re-approval is required) so that we can stop sending excessive email notifications and reduce the footprint of expenses in Freshdesk. 
    2. Platform UX: users can rely on the platform reliably getting their attention where it is needed without feeling tossed around many places on the platform.

### Boundaries

This project creates a space to explore an alternative to the expense-list view - to try something more concise (tabular) vs. the dense-block we currently have in the fiscal host workspace. This has also come up in #6385 as a core UX pattern - switching between an "item blocks view" and a "table view."

This project does and should rely on the existing expense details template. It is placing it in a new container and this may create (justified) pressure to improve the template. Any such improvements are NOT in the scope of this project. Insights about potential improvements should be collected (as assessments) and can be used to propose future improvements to the expense details page.

## Metrics 
The users should feel more confident about the platform and more clear about the work they need to do and their ability to manage it on the platform. When this user experiences coincides with good fiscal host processes (and other platform capabilities some of which we have recently improved and some we may get to in the future) the users should feel they are able to effectively and confidently navigate the work of paying expenses. 

  P1 high frequency, high impact

**Comment by iamronen** [comment]
Kickoff summary

### Overview:
1. The project will receive attention later in the cycle as the team engaged with it is currently occupied with #6385 
2. Part 1 is a clearly framed scope of work that we can take on in this cycle with the intention to release.
3. Part 2 is an understood need but is complex, subtle and needs research input from both design and engineering in order to convert it into an understood solution. We will aspire to advance some of this research during this cycle with the intention to formulate a pitch that can be prioritized in future cycles.

### Part 1: Expense Pipeline Overview
1. The time from now and until we start implementing this gives design an opportunity to come up with alternate proposals (then that  established by the pitch wireframe) to address the needs surfaced in the pitch. 
2. OCF would like to be able to use tags to delineate sets of expenses that are handled by different host admins (eg: all leases are handled by AdminA and all cash assistance programs are handled by AdminB). We acknowledge that tags are a fragile mechanism for achieving this but would like to make this available to OCF by adding filtering by tags.
3. @kewitz surfaced issues that can arise from fiscal hosts admin working in parallel:
     1. Implementing some kind of "record lock" to prevent two fiscal hosts from attempting to pay the same expense. We acknowledged that in the future this is something we may need to address but that it does not pose an immediate concern because very few hosts are actually operating with parallel users making payment and there are already mechanisms proposed to segregate work (eg: one admin working from newest to oldest and another from oldest to newest, or tag-filtering)
     2. How to manage screen updates (to both lists and counters) to reflect in a meaningful way the state of the pipeline. Eg: when an expense is marked as incomplete it should "move" from the "ready-to-pay" section to the "incomplete" section, the ready-to-pay counter should drop by one and the incomplete should grow by 1. 
     3. And, by extension how to do this in a way that account for users working in parallel. Eg: when UserA marks an expense as incomplete the lists and numbers are updated for both UserA and UserB.

### Part 2: Calling for Attention
1. Purpose: the attention indicators, together with the pipeline overview are intended to create a platform experience that users can trust and lean into for managing expenses. This means that all the actors (expense submitters, collective admins and fiscal host admins) involved are reliable notified when there are things they need to do around expenses to keep the pipeline flowing and prevent it from backing up. Getting there means that the majority of the expense flow (especially for large fiscal host admins who work daily on the platform) can be decoupled from email and support channels. 
2. How to implement attention indicators in a way that they are helpful but not intrusive.
3. How to implement attention indicators in a way that users can dismiss them (eg: mark as read).

**Comment by tomsusemihl** [comment]
UX tasks

- [ ] Launch a discovery usability test with the [live-coded prototype with the starter (A) proposal](https://opencollective-frontend-git-tailwind-ui-opencollective.vercel.app/dashboard/opensource/host-expenses?status=READY_TO_PAY)
- [ ] Evaluate and sketch other UX proposals (B, C...) and prepare then for testing

**Comment by tomsusemihl** [comment]
- [ ] Prepare a way to gather user feedback (Related to the whole Workspace feedback campaign)

**Comment by gustavlrsn** [comment]
### This week
- [ ] add tabs (and use them with existing filters) behind a preview feature flag accessible to first party hosts

**Comment by tomsusemihl** [comment]
- [ ] Usability Testing with users

**Comment by tomsusemihl** [comment]
- [ ] Feedback analysis and UX/Copy iteration proposals

**Comment by BenJam** [comment]
not sure about release status, lets pick up with @kewitz later this week.
